client/post-editor/editor-visibility/index.jsx

```diff
@@ -189,6 +189,8 @@ const EditorVisibility = React.createClass( {
 	updateVisibility( newVisibility ) {
 		const { siteId, postId } = this.props;
 		let reduxPostEdits;
+		const defaultVisibility = 'draft' === this.props.status ? 'draft' : 'publish';
+		const postEdits = { status: defaultVisibility };
```

Comment:
This needs a comment. Because I didn't understand what the code was doing when it was originally in here, thought we had regression-tested the related parts of product to confirm things were still working, and thought this was just old code that could be removed. I don't want to see somebody else make the same mistake again!

It would be better to also rework the code to make it clear that it is handling the case from private -> public/password.

Even better, let's consolidate these additions to a single function called from this one.